### PR TITLE
Group client projects under :client

### PR DESCRIPTION
**Pull request by nik9000**
```
:client ---------> :client:low-level
:client-sniffer -> :client:sniffer
:client-test ----> :client:test
```

This lines the client up with how we do things like modules and plugins.

### CODE REVIEW

**Comment by nik9000** [comment]
I think it might also be nice to move the package the files are in to line up with their projects but I'd prefer to wait on that because it'll create a noisy diff.

**Comment by nik9000** [comment]
@javanna and @rjernst what do you think?

**Comment by rjernst** [comment]
I think I would call `low-level` instead `rest`? I believe @s1monw has plans to move the transport client into `:client:transport`

File: settings.gradle [hunk 1]
```diff
@@ -5,9 +5,10 @@ List projects = [
   'rest-api-spec',
   'core',
   'docs',
-  'client',
-  'client-sniffer',
   'client-test',
```
Comment (rjernst) [inline settings.gradle]: Why is this `client-test` still here, in addition to `client:test`?

**Comment by rjernst** [comment]
LGTM, a couple small suggestions.

**Comment by nik9000** [comment]
> I think I would call low-level instead rest

I'm fine with that. It is the "low-level" REST client but we can rename it later if we want.

**Comment by nik9000** [comment]
Thanks for reviewing @rjernst !

**Pull request merged**